ci-operator/step-registry/ovn/defaultnetwork-internal-transit-switch-subnet/ovn-defaultnetwork-internal-transit-switch-subnet-commands.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+set -x
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source "${SHARED_DIR}/ovn-utils.sh"
+
+# make sure cluster is up and healthy after install and dump initial state. There are cases when not
+# all operators are ready even after the install process has completed. Poll for another 15m to be
+# sure and exit/fail if all operators are not healthy.
+wait_for_operators_and_nodes 900
+dump_cluster_state
+
+# modify networks.operator.openshift.io with custom "internalJoinSubnet"
+oc patch networks.operator.openshift.io cluster --type=merge -p '{"spec":{"defaultNetwork":{"ovnKubernetesConfig":{"ipv4":{"internalTransitSwitchSubnet": "100.69.0.0/16"}}}}}'
+
+internalTransitSwitchSubnet=$(oc get network.operator.openshift.io -o jsonpath='{.items[0].spec.defaultNetwork.ovnKubernetesConfig.ipv4.internalTransitSwitchSubnet}')
+
+# first wait for the network operator to move to Progressing=True
+wait_for_operator_to_be_progressing network
+
+# make sure all operators and nodes are good
+wait_for_operators_and_nodes 300
+
+if [[ "$internalTransitSwitchSubnet" != "100.69.0.0/16" ]]; then
+  echo "Error: internalTransitSwitchSubnet is misconfigured. Expected internalTransitSwitchSubnet of 100.69.0.0/16, but got:"
+  oc get network.operator.openshift.io -o jsonpath='{.items[0].spec.defaultNetwork}'
+  exit 1
+fi
```

Comment:
I don't really understand why you check internalTransitSwitchSubnet. Either at all (since it would mean that the patch command that did not fail didn't work, so testing the patch command?) and also why you do it after the operators stabilize but with the value you got before they did.

Reply:
I guess the same comment would apply for the internalJoinSubnet.

Probably missing something that could be clarified through a comment

Reply:
great catch here. the intention was to do the patch, then let the operator to progress and settle, then validate that
it's present in network.opertor.

I just updated the PR to do that with a small comment.

Thanks for this.